test/extended/util/framework.go

```diff
@@ -238,10 +238,55 @@ func processScanError(log string) error {
 	return fmt.Errorf(log)
 }
 
+// WaitForCustomImageStreams creates custom ruby imageStream in OpenShift namespace
+// TODO: To eliminate the dependency on OpenShift Samples Operator in future,
+// WaitForCustomImageStreams should be a replacement of WaitForOpenShiftNamespaceImageStreams func
+func WaitForCustomImageStreams(oc *CLI) error {
+	ctx := context.Background()
+	var registryHostname string
+
+	images := map[string]string{
+		"ruby": "registry.access.redhat.com/ubi8/ruby-33:latest",
+	}
+
+	// Check to see if we have ImageRegistry enabled
+	hasImageRegistry, err := IsCapabilityEnabled(oc, configv1.ClusterVersionCapabilityImageRegistry)
+	if err != nil {
+		return err
+	}
+	if hasImageRegistry {
+		registryHostname, err = WaitForInternalRegistryHostname(oc)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Create custom imageStream using `oc import-image`
+	e2e.Logf("waiting for imagestreams to be imported")
+	for image, location := range images {
+		err := oc.Run("import-image").Args(image, fmt.Sprintf("--from=%s", location), "--confirm").Execute()
+		if err != nil {
+			return err
+		}
+
+		// Wait for imageRegistry to be ready
+		pollErr := wait.PollUntilContextTimeout(ctx, 10*time.Second, 150*time.Second, false, func(context.Context) (bool, error) {
+			return checkOpenShiftNamespaceImageStreamImported(ctx, oc, image, registryHostname)
+		})
+		// pollErr will be not nil if there was an immediate error, or we timed out.
+		if pollErr != nil {
+			return pollErr
+		}
+		DumpImageStream(oc, "openshift", image)
+	}
```

Comment:
@adambkaplan here, I have returned the pollErr immediately w/o calling DumpImageStream(), unlike what we have in [WaitForSamplesImagestream](https://github.com/openshift/origin/blob/master/test/extended/util/framework.go#L306-L309). Should this be fine, or there's any reason we want to call DumpImageStream() & then return the pollErr?